zerver/lib/upload.py

```diff
@@ -130,6 +130,16 @@ def get_realm_for_filename(path):
         return None
     return get_user_profile_by_id(key.metadata["user_profile_id"]).realm.id
 
+def delete_message_image_s3(path_id):
```

Comment:
Cool.  The goal should be to have automated tests for both codepaths.

The tests shouldn't talk to real S3 (since then the tests won't work when not in the internet).

In https://github.com/zulip/zulip/issues/664 I have a proposal for how to do that.  Maybe it makes sense to try to get that proposal working, and then we can use the framework from that proposal to add an automated test for this as well?